test/node/src/safe-empty-where-in.test.ts

```diff
@@ -0,0 +1,916 @@
+        expect(result).to.deep.equal([])
+
+        const notInQuery = db
+          .selectFrom('safeEmptyArrayPerson')
+          .where('firstName', 'not in', [])
+          .select('safeEmptyArrayPerson.firstName')
+
+        testSql(notInQuery, dialect, {
+          postgres: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in ($1)`,
+            ],
+            parameters: [null],
+          },
+          mysql: {
+            sql: [
+              'select `safeEmptyArrayPerson`.`firstName`',
+              'from `safeEmptyArrayPerson`',
+              'where `firstName` not in (?)',
+            ],
+            parameters: [null],
+          },
+          mssql: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (@1)`,
+            ],
+            parameters: [null],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (?)`,
+            ],
+            parameters: [null],
+          },
+        })
+
+        result = await notInQuery.execute()
+
+        expect(result).to.deep.equal([])
+      })
+
+      it('non-empty array select from should return expected results', async () => {
+        const query = db
+          .selectFrom('safeEmptyArrayPerson')
+          .where('firstName', 'in', ['John', 'Mary'])
+          .select('safeEmptyArrayPerson.firstName')
+
+        testSql(query, dialect, {
+          postgres: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in ($1, $2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mysql: {
+            sql: [
+              'select `safeEmptyArrayPerson`.`firstName`',
+              'from `safeEmptyArrayPerson`',
+              'where `firstName` in (?, ?)',
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mssql: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (@1, @2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (?, ?)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+        })
+
+        let result = await query.execute()
+
+        expect(result).to.have.length(2)
+        expect(result).to.deep.equal([
+          { firstName: 'John' },
+          { firstName: 'Mary' },
+        ])
```

Comment:
no need to execute really, testing the compiled sql is more than enough.

Reply:
Could we opt to keep to test for runtime errors